locations/middlewares/anti_bot_detection.py

```diff
@@ -0,0 +1,103 @@
+from enum import Enum
+
+from scrapy import Request, Spider
+from scrapy.downloadermiddlewares.httpcompression import HttpCompressionMiddleware
+from scrapy.http import HtmlResponse, Response
+
+
+# As a temporary transition to use of AntiBotDetectionMiddleware,
+# assume that Zyte API is capable of bypassing all anti-bot
+# methods. This list will need to be amended after testing of Zyte
+# API ability to bypass various anti-bot methods. This list will
+# need to be frequently updated depending on the effectiveness of
+# Zyte API's ability to rapidly bypass new/revised anti-bot methods.
+class AntiBotMethods(Enum):
+    # Azure Web Applicatigon Firewall bot protection documentation:
+    # https://learn.microsoft.com/en-us/azure/web-application-firewall/ag/bot-protection-overview
+    AZURE_WAF = {"name": "Azure WAF", "zyte_bypassable": True}
+
+    # Cloudflare bot protection documentation:
+    # https://developers.cloudflare.com/bots/
+    CLOUDFLARE = {"name": "Cloudflare", "zyte_bypassable": True}
```

Comment:
How true is this? We hit a lot of CF sites already with no issues, we log the stats too iirc. I don't know the CF terminology, but there are different levels, we handle some fine, playwright handles some, but does Zyte work for them all? Or are there limits there too? Does it matter?

Reply:
Generally I've only deemed something to be "anti-bot" if there is a CAPTCHA or JavaScript challenge required to be completed by the client. Cloudflare and many other services have confusing marketing, but some will provide the company with a choice of implementing IP address geographic restrictions, data centre IP address blocking, CAPTCHA challenges, JavaScript challenges, rate limiting, etc.

I don't believe Zyte have a status page available where they advertise their current/live status of where they are in the arms race with anti-bot services.  It therefore seems like we'd have to trial and error approach (perhaps even before each crawl commences, if Scrapy allows this) to populate the `zyte_bypassable` values.